.pre-commit-config.yaml

```diff
@@ -0,0 +1,26 @@
+repos:
+  -   repo: https://github.com/PyCQA/isort
+      rev: 5.6.4
+      hooks:
+      - id: isort
+  -   repo: https://gitlab.com/pycqa/flake8
+      rev: 3.8.4
+      hooks:
+      - id: flake8
+  -   repo: https://github.com/pre-commit/pre-commit-hooks
+      rev: v3.3.0
+      hooks:
+      -   id: trailing-whitespace
+      -   id: end-of-file-fixer
+      -   id: check-case-conflict
+      -   id: check-json
+  -   repo: https://github.com/pre-commit/mirrors-eslint
+      rev: '7.15.0'
+      hooks:
+      -   id: eslint
+  -   repo: https://github.com/pre-commit/pygrep-hooks
+      rev: v1.7.0
+      hooks:
+      -   id: rst-backticks
```

Comment:
Ha fun, I did a whole thing to block these in #12645 .

One thing about these rst hooks is they use `types: [rst]` - our rst uses the `.txt` extension, did you check they run on our docs correctly?

Reply:
Thanks for this. 

It's a bit intermittent, some  work, some don't. I'll ask if we can make them all work.

Reply:
I've added some additional configuration options so this now works for `.rst` and `.txt` files. 

```
rst ``code`` is two backticks............................................Failed
- hook id: rst-backticks
- exit code: 1

test.rst:1:a `failing` file:
test.txt:1:a `failing` file:

rst directives end with two colons.......................................Failed
- hook id: rst-directive-colons
- exit code: 1

test.rst:6:.. versionadded:
test.txt:6:.. versionadded:
```

Reply:
However, this now captures other text files that are in the repo which we don't want. :disappointed: